Error compiling Windows DLL

Until recently I was able to successfully compile libstorj to a Windows DLL following these instructions: https://github.com/Storj/libstorj#cross-compiling-dependencies-from-ubuntu-1604

I pulled from master today and tried to build again, but I've got the following errors:

```
cli_callback.c: In function ‘printdir’:
cli_callback.c:63:5: error: unknown type name ‘DIR’
     DIR *dp;
     ^
cli_callback.c:73:15: warning: implicit declaration of function ‘opendir’ [-Wimplicit-function-declaration]
     if ((dp = opendir(dir)) == NULL) {
               ^
cli_callback.c:73:13: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
     if ((dp = opendir(dir)) == NULL) {
             ^
cli_callback.c:79:21: warning: implicit declaration of function ‘readdir’ [-Wimplicit-function-declaration]
     while ((entry = readdir(dp)) != NULL) {
                     ^
cli_callback.c:79:19: warning: assignment makes pointer from integer without a cast [-Wint-conversion]
     while ((entry = readdir(dp)) != NULL) {
                   ^
cli_callback.c:80:9: warning: implicit declaration of function ‘lstat’ [-Wimplicit-function-declaration]
         lstat(entry->d_name, &statbuf);
         ^
cli_callback.c:80:20: error: dereferencing pointer to incomplete type ‘struct dirent’
         lstat(entry->d_name, &statbuf);
                    ^
cli_callback.c:89:25: warning: implicit declaration of function ‘realpath’ [-Wimplicit-function-declaration]
             full_path = realpath(entry->d_name, NULL);
                         ^
cli_callback.c:95:5: warning: implicit declaration of function ‘closedir’ [-Wimplicit-function-declaration]
     closedir(dp);
     ^
cli_callback.c: In function ‘file_exists’:
cli_callback.c:102:5: error: unknown type name ‘gid_t’
     gid_t  st_grpid;
     ^
cli_callback.c:136:14: error: ‘S_IFSOCK’ undeclared (first use in this function)
         case S_IFSOCK:
              ^
cli_callback.c:136:14: note: each undeclared identifier is reported only once for each function it appears in
Makefile:525: recipe for target 'cli_callback.lo' failed
```

Is there anything missing in the instructions that has to be added for the latest changes?

--- a/src/cli.c
+++ b/src/cli.c
@@ -113,9 +113,11 @@ static int check_file_path(char *file_path)
         case S_IFREG:
             return CLI_VALID_REGULAR_FILE;
             break;
+#ifdef S_IFSOCK
         case S_IFSOCK:
             printf("socket\n");
             break;
+#endif
         default:
             printf("unknown?\n");
             break;

--- a/src/cli_callback.c
+++ b/src/cli_callback.c
@@ -1,4 +1,5 @@
 #include "cli_callback.h"
+#include <dirent.h>
 
 //#define debug_enable
 
@@ -63,12 +64,6 @@ static void printdir(char *dir, int depth, FILE *src_fd, void *handle)
     DIR *dp;
     struct dirent *entry;
     struct stat statbuf;
-    int spaces = depth*4;
-    char tmp_dir[800] = {};
-    char *full_path = NULL;
-    char *s;
-    char * start;
-    cli_api_t *cli_api = handle;
 
     if ((dp = opendir(dir)) == NULL) {
         fprintf(stderr,"cannot open directory: %s\n", dir);
@@ -77,7 +72,7 @@ static void printdir(char *dir, int depth, FILE *src_fd, void *handle)
 
     int ret = chdir(dir);
     while ((entry = readdir(dp)) != NULL) {
-        lstat(entry->d_name, &statbuf);
+        stat(entry->d_name, &statbuf);
         if (S_ISDIR(statbuf.st_mode)) {
             /* Found a directory, but ignore . and .. */
             if (strcmp(".", entry->d_name) == 0 ||
@@ -86,20 +81,17 @@ static void printdir(char *dir, int depth, FILE *src_fd, void *handle)
             /* Recurse at a new indent level */
             printdir(entry->d_name, depth + 1, src_fd, handle);
         } else {
-            full_path = realpath(entry->d_name, NULL);
             /* write to src file */
-            fprintf(src_fd, "%s%s\n", "", full_path);
+            fprintf(src_fd, "%s%s\n", "", entry->d_name);
         }
     }
     ret = chdir("..");
     closedir(dp);
-    free(full_path);
 }
 
 static int file_exists(void *handle)
 {
     struct stat sb;
-    gid_t  st_grpid;
     cli_api_t *cli_api = handle;
 
     FILE *src_fd, *dst_fd;
@@ -133,9 +125,11 @@ static int file_exists(void *handle)
         case S_IFREG:
             return CLI_VALID_REGULAR_FILE;
             break;
+#ifdef S_IFSOCK
         case S_IFSOCK:
             printf("socket\n");
             break;
+#endif
         default:
             printf("unknown?\n");
             break;

